lib/telnet.c

```diff
@@ -1172,20 +1172,20 @@ static CURLcode send_telnet_data(struct Curl_easy *data,
                                  char *buffer, ssize_t nread)
 {
   ssize_t escapes, i, outlen;
-  unsigned char *outbuf = NULL;
+  char *outbuf = NULL;
```

Comment:
This change *adds* to the total number of typecasts, so why change it?

Reply:
The main issue here is that `send_telnet_data` accepts a `char *buffer` not an `unsigned char *buffer`. So one way to remove all the casts is to change the interface. Would you accept such a change?

Reply:
If it cleans up things, sure!